lib/collection/src/shards/shard_holder/resharding.rs

```diff
@@ -220,16 +220,20 @@ impl ShardHolder {
     pub fn check_abort_resharding(&mut self, resharding_key: &ReshardKey) -> CollectionResult<()> {
         let state = self.resharding_state.read();
 
-        // `abort_resharding` designed to be self-healing, so...
-        //
-        // - it's safe to run, if resharding is *not* in progress
+        // - do not abort if no resharding operation is ongoing
         let Some(state) = state.deref() else {
-            return Ok(());
+            return Err(CollectionError::bad_request(format!(
+                "can't abort resharding {resharding_key}, no resharding operation in progress",
+            )));
         };
 
-        // - it's safe to run, if *another* resharding is in progress
+        // - do not abort if there is no active reshardinog operation with that key
         if !state.matches(resharding_key) {
-            return Ok(());
+            return Err(CollectionError::bad_request(format!(
+                "can't abort resharding {resharding_key}, \
+                 resharding operation in progress has key {}",
+                state.key(),
+            )));
         }
```

Comment:
Not 100% sure about this one. The idea was that abort is self-healing, and so `check_abort_resharding` allows abort even if resharding is not in progress or if a key does not match, so that abort can follow through and potentially repair any left-over state. If resharding for provided key is not in progress and state is correct, it would just be a no-op.

Reply:
> If resharding for provided key is not in progress and state is correct, it would just be a no-op.

Sadly this appears not to be the case. 

For example, this will apply subtraction even if we expect it to be a no-op: https://github.com/qdrant/qdrant/blob/78041dcf0ab4fcb86950d3d967a079ebeb015838/lib/collection/src/collection/resharding.rs#L213-L215

We might want to adjust that instead.